### PR TITLE
let attack like vanilla

### PR DESCRIPTION
**Pull request by DylanMarkon**
## Introduction
let attack like vanilla
### Relevant issues
none

* Fixes #1 hit entity with break sprinting

### API changes
none
### Behavioural changes
hit entity with break sprinting
## Backwards compatibility
none

## Follow-up
none
Requires translations:
none
## Tests
tested , and its works :).

### CODE REVIEW

**Comment by TwistedAsylumMC** [comment]
How does this relate at all to the mentioned pull request?

**Comment by Blackjack200** [comment]
nice

**Comment by dktapps** [comment]
This breaks the critical hit check just a few lines below, and it also shouldn't take effect if the event is cancelled.

TL;DR: this is in the wrong place.

**Comment by DylanMarkon** [comment]
> This breaks the critical hit check just a few lines below, and it also shouldn't take effect if the event is cancelled.
> 
> TL;DR: this is in the wrong place.

kk I fixed it in the recently commit

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -2619,7 +2619,7 @@ public function handleInventoryTransaction(InventoryTransactionPacket $packet) :
 								$cancelled = true;
 							}
 						}
-
+
```
Comment (dktapps) [inline src/pocketmine/Player.php:2622]: please remove this change

**Comment by Encritary** [comment]
This might be an unpleasant change for PvP servers, so it's better to add an option to disable this through API (maybe ``EntityDamageByEntityEvent::setSprintBreak(false)``)

**Comment by dktapps** [comment]
oh noes

**Comment by dktapps** [comment]
Something tells me half of the PvP community is going to complain about this change. The issue is that it's possible to implement this using a plugin right now without any core changes, but not possible to stop it being activated if it becomes core behaviour.

**Comment by Blackjack200** [comment]
> Something tells me half of the PvP community is going to complain about this change. The issue is that it's possible to implement this using a plugin right now without any core changes, but not possible to stop it being activated if it becomes core behaviour.

i will add a function to handle sprint

**Review by BenHeroPH — APPROVED** [review]
this is good.

**Comment by ghost** [comment]
> This might be an unpleasant change for PvP servers, so it's better to add an option to disable this through API (maybe ``EntityDamageByEntityEvent::setSprintBreak(false)``)

A developer can just use ``Player::setSprinting(false)``
